### PR TITLE
Improvement/contentful 0.7.1

### PR DESCRIPTION
**Pull request by loudmouth**
Upgrade and refactor to use [contentful.swift 0.7.1](https://github.com/contentful/contentful.swift/releases/tag/0.7.1). The refactor:
- Flips the dependency relationship between contentful.swift and contentful-persistence.swift
- Now the main SDK sends messages to types conforming to `PersistenceIntegration`

Other changes
- Protocol names and required variables/methods changed for more clarity.
- Changelog added
- Scripted manner for versioning the sdk
- Make command for setting up developer environment to make contributing easier

Fixes: #20 #19

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 language: objective-c
 osx_image: xcode8.3
 rvm:
-  - 2.3.0
+  - 2.4.0
```
Comment (dlitvakb) [inline .travis.yml]: Should be updated to 2.4.1, there are many bugs in 2.4.0 - which may cause your builds to randomly fail.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
+This project adheres to [Semantic Versioning](http://semver.org/) starting from 1.x releases.
+
+### Merged, but not yet released
+> #### Changed
+> **BREAKING:** `ContentfulSynchronizer` is now called `SynchronizationManager`
+> **BREAKING:** Rather than initializing `SynchronizationManager` with a `Contentful.Client` instance, the `Contentful.Client` is now initialized with a `SynchronizationManager` instance as the `persistenceIntegration` parameter in the `Client` initializer.
+> **BREAKING:** The manner in which content type identifiers map Contentful responses to `NSManagedObject` model classes is now changed: `SynchronizationManager` is initialized with a `PersistenceModel` which is constructed
```
Comment (dlitvakb) [inline CHANGELOG.md]: Is there something missing here? I don't understand the end of the sentence.

Comment (loudmouth) [inline CHANGELOG.md]: @dlitvakb just pushed the end of the sentence ;-)

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,7 +7,8 @@ This project adheres to [Semantic Versioning](http://semver.org/) starting from
 > #### Changed
 > **BREAKING:** `ContentfulSynchronizer` is now called `SynchronizationManager`
 > **BREAKING:** Rather than initializing `SynchronizationManager` with a `Contentful.Client` instance, the `Contentful.Client` is now initialized with a `SynchronizationManager` instance as the `persistenceIntegration` parameter in the `Client` initializer.
-> **BREAKING:** The manner in which content type identifiers map Contentful responses to `NSManagedObject` model classes is now changed: `SynchronizationManager` is initialized with a `PersistenceModel` which is constructed
+> **BREAKING:** The manner in which content type identifiers map Contentful responses to `NSManagedObject` model classes is now changed: `SynchronizationManager` is initialized with a `PersistenceModel` which is constructed by passing in your `NSManagedObject` subclasses that conform to either `SyncSpacePersistable` `AssetPersistable` or `EntryPersistable`.
```
Comment (dlitvakb) [inline CHANGELOG.md]: There's a missing comma between `SyncSpacePersistable` and `AssetPersistable`

**Pull request merged**